### PR TITLE
Move in block editor exclusive extension

### PR DESCRIPTION
**Pull request by azrikahar**
Closes ENG-939

### CODE REVIEW

File: .changeset/ten-wasps-relate.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@directus/app': minor
+---
+
+Add block editor interface
```
[the remark this answers is not in the document]
Comment (azrikahar) [reply .changeset/ten-wasps-relate.md:5]: Thanks! 😄

**Comment by jaads** [comment]
I'm getting the error `this.ui is undefined` on the console when trying to add an image in the block editor 🤔

**Comment by jaads** [comment]
funny bug: when I add a codeblock with something like a text block before the code block and I press the return key, the text within the text block gets deleted instead of the text in the code block.

https://github.com/directus/directus/assets/34009329/9bb3985d-272a-4790-ada9-39b495e135cb

**Review by jaads — CHANGES_REQUESTED** [review]
see comment above

**Review by jaads — APPROVED** [review]

**Comment by jaads** [comment]
Now it works 👍

**Review by licitdev — COMMENTED** [review]
Adding background info for https://github.com/directus/directus/pull/18525/commits/ca32fbcb6ab669cb6cfc8decd0d42cec90fba63c

Override the blue colour with primary colour and fix delete confirmation button in light mode.


| Mode  | Before  | After  |
|---|---|---|
| Light  | <video src="https://github.com/directus/directus/assets/26413686/9658cda5-98a7-4c83-b74c-1e60b7a416ae">  | <video src="https://github.com/directus/directus/assets/26413686/e64bc877-bd33-4685-9161-1cbd6ab5bffa">  |
| Dark  | <video src="https://github.com/directus/directus/assets/26413686/6bf25dbd-88bf-418a-a4ca-da98ab44342b">  | <video src="https://github.com/directus/directus/assets/26413686/7d29a83b-d71b-4400-8249-adc4e99fa7be">  |

As for https://github.com/directus/directus/pull/18525/commits/ad094af944cb04c86dabc5a5d2c177d7ae04b5c3, the red background is added to be consistent with the demo in https://editorjs.io/.

Selected `--danger-alt` instead of `--danger` as it looks too "bright" in dark mode and requires a font color change to white for light mode.

| Type  | Screenshot  |
|---|---|
| Light  | image  |
| Dark  | image  |
| Official EditorJS  | image  |

**Comment by LasseRosenow** [comment]
Will this integration allow us to use editor.js as a M2A interface for example?

So it is possible to create custom blocks based on directus relations?

If not .. Is this a planned feature?

File: app/src/interfaces/input-block-editor/input-block-editor.vue [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+	if (props.disabled || !context || !context.saver) return;
+	haveValuesChanged.value = true;
+
+	try {
+		const result: EditorJS.OutputData = await context.saver.save();
+
+		if (!result || result.blocks.length < 1) {
+			emit('input', props.nullable ? null : '{}');
+			return;
+		}
```
Comment (licitdev) [inline app/src/interfaces/input-block-editor/input-block-editor.vue:159]:
```suggestion
		if (!result || result.blocks.length < 1) {
			emit('input', null);
			return;
		}
```

Can we remove the `nullable` prop? 

The returning of `"{}"` results in type warnings as the type for `value` is `Record<string, any> | null`.

**Review by licitdev — APPROVED** [review]
LGTM apart from the open conversation to fix type warning.

There's the https://github.com/kommitters/editorjs-toggle-block/issues/176 issue which currently affects the opening of settings popover for the toggle block. In addition, the same error is thrown when the `tab` key is pressed, even when in a different interface such as `Code`.

File: app/src/interfaces/input-block-editor/input-block-editor.vue (hunk 2 above)
Comment (azrikahar) [reply app/src/interfaces/input-block-editor/input-block-editor.vue:159]: I think this makes sense 👍 Will proceed to implement this.

**Comment by LasseRosenow** [comment]
> @LasseRosenow This first iteration is going to be just the "regular" JSON structure. That being said, we are about to dive into improving the M2A builder and potentially also creating an editorjs-type setup that works with that data model (although that is quite a bit trickier) 🤞🏻

That sounds amazing :) 
Really looking forward towards an improved M2A builder :)

**Comment by JohannesPertl** [comment]
I can't find any info in the documentation. Is it possible to add my own collections to the block editor?

**Comment by paescuj** [comment]
Please follow https://docs.directus.io/getting-started/support.html#community-support for questions & help. Thank you!

**Pull request merged**